.github/workflows/run_pr_tests.yml

```diff
@@ -0,0 +1,84 @@
+# Simple workflow for building sccache for PR testing
+name: Run ock tests for PR testing
+# Note this will currently create a new sscache file and this must be manually pruned until such time as we have a job
+# for this.
+on:
+  pull_request:
+    paths:
+      - 'source/**'
+      - 'module/**'
+      - 'examples/**'
+      - 'cmake/**'
+      - 'hal/**'
+      - '.github/actions/do_build_ock/**'
+      - '.github/actions/setup_ubuntu_build/**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+
+  # build and run host x86_64, execute UnitCL and lit tests amd build and run offline
+  run_host_x86_64:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      
+      # installs tools, ninja and installs llvm (default 16, RelAssert)
+      - name: setup-ubuntu
+        uses:  ./.github/actions/setup_ubuntu_build
+  
+      # These need to match the configurations of build_mr_cache to use the cache effectively
+      - name: build_host_x86_64-online-release
+        uses: ./.github/actions/do_build_ock
+        with:
+          build_type: Release
+
+      - name: run host online check
+        run:
+          ninja -C build check-UnitCL
+
+      # use the previous build for online to get clc
+      - name: build_host_x86_64-offline
+        uses: ./.github/actions/do_build_ock
+        with:
+          build_type: Release
+          extra_flags: -DCA_RUNTIME_COMPILER_ENABLED=OFF -DCA_EXTERNAL_CLC=${{ github.workspace }}/build/bin/clc
+          build_dir: build_offline
+          build_targets: UnitCL
+
+      - name: run host_x86 offline
+        run:
+          ninja -C build_offline check-UnitCL
+
+      - name: run just online lit
+        run:
+          ninja -C build check-all-lit
+
+
+
+  # build and run riscv m1, execute UnitCL and lit tests
+  run_riscv_m1:  
+
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      
+      # installs tools, ninja and installs llvm (default 16, RelAssert)
```

Comment:
Do we need to say `(default 16, RelAssert)` all the time? Would this be better left as being explicitly set (using `with:`?)

Otherwise I fear these comments will easily become stale.

Reply:
maybe, I was trying to keep the file short